### PR TITLE
fix some issues in the convective bl model and les

### PR DESCRIPTION
**Pull request by yairchn**
### Description
The convective BK model and LES were not well defined as dry model. This PR fixes these issues 

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: experiments/AtmosLES/convective_bl_model.jl [hunk 1]
```diff
@@ -283,9 +317,15 @@ function convective_bl_model(
                 moisture = moisture_bc,
                 turbconv = turbconv_bcs(turbconv)[1],
             ),
+            AtmosBC(),
             AtmosBC(turbconv = turbconv_bcs(turbconv)[2]),
-        ),
+        )
+    end
+
+    moisture_flux = FT(0)
```
Comment (charleskawczynski) [inline experiments/AtmosLES/convective_bl_model.jl:325]:
```suggestion
```

File: experiments/AtmosLES/convective_bl_model.jl [hunk 2]
```diff
@@ -283,9 +317,15 @@ function convective_bl_model(
                 moisture = moisture_bc,
                 turbconv = turbconv_bcs(turbconv)[1],
             ),
+            AtmosBC(),
```
Comment (charleskawczynski) [inline experiments/AtmosLES/convective_bl_model.jl:320]:
```suggestion
```
I'm not sure what it means to have three `AtmosBC`'s in this context. Should we delete this?

File: experiments/AtmosLES/convective_bl_model.jl [hunk 3]
```diff
@@ -271,8 +291,22 @@ function convective_bl_model(
...
+            AtmosBC(
+                momentum = Impenetrable(DragLaw(
+                    # normPu_int is the internal horizontal speed
+                    # P represents the projection onto the horizontal
+                    (state, aux, t, normPu_int) -> (u_star / normPu_int)^2,
+                )),
+                energy = energy_bc,
+                turbconv = turbconv_bcs(turbconv)[1],
+            ),
+            AtmosBC(),
```
Comment (charleskawczynski) [inline experiments/AtmosLES/convective_bl_model.jl:305]:
```suggestion
```
?

File: experiments/AtmosLES/convective_bl_model.jl (hunk 1 above)
Comment (yairchn) [reply experiments/AtmosLES/convective_bl_model.jl]: I have move this to line 230, as it will be used of the surface flux scheme is set to "Prescribed"

**Review by charleskawczynski — APPROVED** [review]
LGTM.

**Review by ilopezgp — APPROVED** [review]

File: experiments/AtmosLES/convective_bl_les.jl [hunk 4]
```diff
@@ -68,12 +68,7 @@ function main()
     dgn_config = config_diagnostics(driver_config)
 
     cbtmarfilter = GenericCallbacks.EveryXSimulationSteps(1) do
-        Filters.apply!(
-            solver_config.Q,
-            ("moisture.ρq_tot",),
-            solver_config.dg.grid,
-            TMARFilter(),
-        )
+        Filters.apply!(solver_config.Q, (), solver_config.dg.grid, TMARFilter())
```
Comment (charleskawczynski) [inline experiments/AtmosLES/convective_bl_les.jl:71]: Wait, why are we calling this with an empty tuple? We should remove it if filtering is unnecessary.

**Pull request closed without merging**